frontend/packages/dev-console/src/components/import/deployImage-submit-utils.ts

```diff
@@ -252,6 +252,9 @@ export const createOrUpdateDeployment = (
           ],
         },
       },
+      ...(originalDeployment?.spec.strategy && {
+        strategy: originalDeployment?.spec.strategy,
+      }),
```

Comment:
@debsmita1 If you notice we are not making any changes based on the `originalResource` while creating the object for the `newResource` as these utilities are used for both create and edit flows and to keep the `newResource` object clean and also as Christian mentioned above to avoid accidental overrides to any other properties. You should make the changes in the mergeData function which is used to merge the old and new resources and also in cases where we want to keep some properties of old resource. If you see in mergeData function we already have a check 
```
if (mergedData?.spec?.strategy) {
    mergedData.spec.strategy = newResource.spec.strategy;
  }
```
 for your usecase you just need to change this check to something like
```
if (mergedData?.spec?.hasOwnProperty('strategy')) { 
    mergedData.spec.strategy = newResource.spec?.strategy ?? originalResource?.spec?.strategy; 
}
```

Reply:
Is the point of these `if` conditions in `mergeData` to intentionally override the merged values?
It seems a bit odd that the condition checks the merged state instead of the `newResource` state before entering the condition. Why is that?

Reply:
The usecase I can think of where we need to check the `mergedData` values is when the `originalResource` has a property but during edit that property was removed but when we merged the two objects the property again got added and therefore checking the `mergedData` if it has that property we can update it with `newResource` property value which in this case will be undefined. Although for this usecase where strategy for the new resource (d/dc) is always going to be undefined, the merged data will have the original resource strategy which is required here and we can change this check to 
```
 if (newResource.spec?.hasOwnProperty('strategy')) { 
    mergedData.spec.strategy = newResource.spec.strategy; 
}
```
to update the strategy with newResource strategy value in case of buildStrategy which can be updated using the edit flows.

Reply:
@divyanshiGupta when editing the deployment (which has Recreate strategy), the strategy property is not added in the new Resource obj that we create, and the existing mergeData util reads from the `newResource.spec.strategy` which doesn't exist as a result the merged resource has spec.strategy as undefined. So according to me we must consider reading from the original resource. WDYT ?

Reply:
@debsmita1 as I mentioned above since strategy is going to be undefined in the new resource so during merge strategy prop of original resource will be preserved. You can also try that out. We need to add check on new resource since if we have strategy in it we want the mergedData to have the new strategy in case of buildStrategy as it can be updated using edit flows unlike d/dc strategy in which case the new resource strategy is always going to be undefined and hence the original strategy will be used.